scripts/build/gen_relocate_app.py

```diff
@@ -501,22 +506,21 @@ def get_obj_filename(searchpath, filename):
 
 
 # Extracts all possible components for the input string:
-# <mem_region>[\ :program_header]:<flag_1>[;<flag_2>...]:<file_1>[;<file_2>...]
-# Returns a 4-tuple with them: (mem_region, program_header, flags, files)
+# <mem_region>[:PHDR=program_header][:LMEM=load_memory]:FLAGS=<flag_1>[;<flag_2>...]:FILES=<file_1>[;<file_2>...]
+# Returns a 5-tuple with them: (mem_region, program_header, load_mem, flags, files)
 # If no `program_header` is defined, returns an empty string
 def parse_input_string(line):
-    # Be careful when splitting by : to avoid breaking absolute paths on Windows
-    mem_region, rest = line.split(':', 1)
 
-    phdr = ''
-    if mem_region.endswith(' '):
-        mem_region = mem_region.rstrip()
-        phdr, rest = rest.split(':', 1)
+    line_split = line.split(':')
+    arg_dict = {i.split('=')[0]: i.split('=')[1] for i in line_split[1:]}
```

Comment:
afaict this breaks on Windows where paths contains `:`, such as `c:/`.

Please also take a look at this comment: https://github.com/zephyrproject-rtos/zephyr/pull/74402#discussion_r1715420319

Perhaps you didn't notice the comment you removed here: https://github.com/zephyrproject-rtos/zephyr/blob/c710f8892beef3e486dded2f0ad4394e4a4d9aa0/scripts/build/gen_relocate_app.py#L508